sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c

```diff
@@ -130,8 +224,13 @@ static bool _remove_old_prefix(gnrc_netif_t *netif,
     return false;
 }
 
-static void _configure_subnets(uint8_t subnets, gnrc_netif_t *upstream,
-                               const ndp_opt_pi_t *pio)
+static void _configure_subnets(uint8_t subnets, uint8_t start_idx, gnrc_netif_t *upstream,
+#if IS_USED(MODULE_GNRC_IPV6_AUTO_SUBNETS_SIMPLE)
+                               const ndp_opt_pi_t *pio
+#else
+                               const struct pio_cache *pio
+#endif
```

Comment:
Please don't do it this way. I know I did that too in the NIB, but it is horrible. It could be a nice solution to inherit `struct pio_cache` from `ndp_opt_pi_t` and only add additional fields needed when the pseudo-module is compiled in. If that is too wasteful for your liking, I would prefer different functions (or at least different _complete_ function declarations) for both modules. That way you also don't have some weird typing issues that might show up later, because the type within the function depends on the compile-time config.

Reply:
You're right, when considering padding I'm only saving 4 bytes with the custom `struct pio_cache`.
The added ROM cost of manually copying the struct elements would outweigh that.